### DISCUSSION TITLE
Why not upstream CPM to cmake?

### DISCUSSION THREAD

**Discussion by ghost** [cpm-cmake/CPM.cmake#254] [General]
Currently CPM has features that CMake's FetchContent doesn't.

This issue highlights the disparity pretty well:
https://discourse.cmake.org/t/how-to-handle-duplicate-targets-add-subdirectory-only/3362/6

Issues like this seems like they could be fixed by CPM upstreaming the changes to CMake.

Thoughts?

**Comment by TheLartians** [comment]
Hey @jpr89 , thanks for the suggestion, I would personally love to see this project become an official part of CMake! On the other hand, I also realise that the current modular approach allow us to add and release new features faster and also support different CMake versions.

I'm not affiliated with the CMake team in any way though, so perhaps @craigscott-crascit can offer some additional perspective. :)
